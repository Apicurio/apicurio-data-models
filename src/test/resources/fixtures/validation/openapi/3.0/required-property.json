{
  "openapi": "3.0.0",
  "info": {
    "license": {
    }
  },
  "servers": [
    {
      "description": "srv-001"
    },
    {
      "url": "https://{username}.gigantic-server.com/",
      "description": "svar-001",
      "variables": {
        "username": {
          "description": "this value is assigned by the service provider, in this example `gigantic-server.com`"
        }
      }
    }
  ],
  "paths": {
    "/op-007": {
      "summary": "Foo Resource",
      "description": "CRUD for Foos.",
      "get": {
        "tags": [ "tag1", "tag2" ],
        "summary": "List Pets",
        "description": "Returns all pets from the system that the user has access to",
        "externalDocs": {
          "description": "Find more info here",
          "url": "https://example.com"
        },
        "operationId": "listPets",
        "deprecated": false
      }
    },
    "/par-001": {
      "parameters": [
        {
          "in": "query",
          "description": "username to fetch",
          "required": true,
          "schema": {
            "type": "string"
          }
        }
      ]
    },
    "/par-002": {
      "parameters": [
        {
          "name": "Foo",
          "required": true,
          "schema": {
            "type": "string"
          }
        }
      ]
    },
    "/res-001": {
      "get": {
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/disc-001"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "disc-001": {
        "oneOf": [
          {
            "$ref": "https://gigantic-server.com/schemas/Cat/schema.json"
          },
          {
            "$ref": "https://gigantic-server.com/schemas/Monster/schema.json"
          }
        ],
        "discriminator": {
          "mapping": {
            "dog": "#/components/schemas/Dog",
            "monster": "https://gigantic-server.com/schemas/Monster/schema.json"
          }
        }
      },
      "ed-001": {
        "type": "string",
        "externalDocs": {
          "description": "More info"
        }
      }
    },
    "securitySchemes": {
      "flow-001": {
        "type": "oauth2",
        "flows": {
          "implicit": {
            "tokenUrl": "http://www.tokens.io/token",
            "scopes": {
              "write:pets": "modify pets in your account",
              "read:pets": "read your pets"
            }
          },
          "authorizationCode": {
            "tokenUrl": "http://www.tokens.io/token",
            "scopes": {
              "read:pets": "read your pets"
            }
          }
        }
      },
      "flow-002": {
        "type": "oauth2",
        "flows": {
          "implicit": {
            "authorizationUrl": "http://www.tokens.io/auth",
            "scopes": {
              "write:pets": "modify pets in your account",
              "read:pets": "read your pets"
            }
          },
          "authorizationCode": {
            "authorizationUrl": "http://www.tokens.io/auth",
            "scopes": {
              "read:pets": "read your pets"
            }
          },
          "clientCredentials": {
            "authorizationUrl": "http://www.tokens.io/auth",
            "scopes": {
              "read:pets": "read your pets"
            }
          }
        }
      },
      "flow-006": {
        "type": "oauth2",
        "flows": {
          "implicit": {
            "tokenUrl": "http://www.tokens.io/token",
            "authorizationUrl": "http://www.tokens.io/auth"
          }
        }
      },
      "ss-001": {},
      "ss-002": {
        "type": "apiKey",
        "in": "header"
      },
      "ss-003": {
        "type": "apiKey",
        "name": "api_key"
      },
      "ss-019": {
        "type": "http"
      },
      "ss-020": {
        "type": "oauth2"
      },
      "ss-021": {
        "type": "openIdConnect"
      }
    },
    "requestBodies": {
      "rb-002": {
        "description": "The request body."
      }
    }
  },
  "tags": [
    {
      "description": "tag-001"
    }
  ]
}
