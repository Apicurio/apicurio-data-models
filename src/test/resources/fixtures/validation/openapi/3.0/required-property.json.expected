[INF-001] |medium| {/info->title} :: API is missing a title.
[INF-002] |medium| {/info->version} :: API is missing a version.
[LIC-001] |medium| {/info/license->name} :: License is missing a name.
[TAG-001] |medium| {/tags[0]->name} :: Tag is missing a name.
[OP-007] |medium| {/paths[/op-007]/get->responses} :: Operation must have at least one response.
[PAR-001] |medium| {/paths[/par-001]/parameters[0]->name} :: Parameter is missing a name.
[PAR-002] |medium| {/paths[/par-002]/parameters[0]->in} :: Parameter is missing a location (Query, Header, etc).
[RES-001] |medium| {/paths[/res-001]/get/responses[200]->description} :: Response (code 200) is missing a description.
[SRV-001] |medium| {/servers[0]->url} :: Server is missing a template URL.
[SVAR-001] |medium| {/servers[1]/variables[username]->default} :: Server Variable "username" is missing a default value.
[DISC-001] |medium| {/components/schemas[disc-001]/discriminator->propertyName} :: Discriminator must indicate a property (by name).
[ED-001] |medium| {/components/schemas[ed-001]/externalDocs->url} :: External Documentation is missing a URL.
[FLOW-001] |medium| {/components/securitySchemes[flow-001]/flows/implicit->authorizationUrl} :: Implicit OAuth Flow is missing an Authorization URL.
[FLOW-001] |medium| {/components/securitySchemes[flow-001]/flows/authorizationCode->authorizationUrl} :: Auth Code OAuth Flow is missing an Authorization URL.
[FLOW-002] |medium| {/components/securitySchemes[flow-002]/flows/clientCredentials->tokenUrl} :: Client Credentials OAuth Flow is missing a Token URL.
[FLOW-002] |medium| {/components/securitySchemes[flow-002]/flows/authorizationCode->tokenUrl} :: Auth Code OAuth Flow is missing a Token URL.
[FLOW-006] |medium| {/components/securitySchemes[flow-006]/flows/implicit->scopes} :: OAuth Flow is missing defined scopes.
[SS-001] |medium| {/components/securitySchemes[ss-001]->type} :: Security Scheme is missing a type.
[SS-002] |medium| {/components/securitySchemes[ss-002]->name} :: API Key Security Scheme is missing a parameter name (e.g. name of a header or query param).
[SS-003] |medium| {/components/securitySchemes[ss-003]->in} :: API Key Security Scheme must describe where the Key can be found (e.g. header, query param, etc).
[SS-019] |medium| {/components/securitySchemes[ss-019]->scheme} :: HTTP Security Scheme is missing a scheme (Basic, Digest, etc).
[SS-020] |medium| {/components/securitySchemes[ss-020]->flows} :: OAuth Security Scheme does not define any OAuth flows.
[SS-021] |medium| {/components/securitySchemes[ss-021]->openIdConnectUrl} :: OpenID Connect Security Scheme is missing a Connect URL.